
<launch>

    <arg name="remove_ground" default="true" />
    <arg name="downsample_cloud" default="true" /> <!-- Apply VoxelGrid Filter with the value given by "leaf_size"-->
    <arg name="leaf_size" default="0.5" /><!-- Voxel Grid Filter leaf size-->
    <arg name="sync" default="false" />
    <arg name="use_diffnormals" default="false" />
    <arg name="pose_estimation" default="true" />
    <arg name="clip_min_height" default="-2" />
    <arg name="clip_max_height" default="1" />
    <arg name="keep_lanes" default="false" />
    <arg name="keep_lane_left_distance" default="15" />
    <arg name="keep_lane_right_distance" default="15" />
    <arg name="max_boundingbox_side" default="10" />
    <arg name="cluster_merge_threshold" default="1.5" />
    <arg name="clustering_distance" default="0.75" />
    <arg name="use_vector_map" default="false" />
    <arg name="wayarea_gridmap_layer" default="wayarea" />
    <arg name="output_frame" default="velodyne" />
    <arg name="remove_points_upto" default="5" />

      
    <node pkg="floor_segmentation" type="floor_segmentation" name="floor_segmentation" output="screen">

    <param name="remove_ground" value="$(arg remove_ground)"/>
    <param name="downsample_cloud" value="$(arg downsample_cloud)"/>
    <param name="leaf_size" value="$(arg leaf_size)"/>
    <param name="use_diffnormals" value="$(arg use_diffnormals)"/>
    <param name="pose_estimation" value="$(arg pose_estimation)"/>
    <param name="keep_lanes" value="$(arg keep_lanes)"/>
    <param name="keep_lane_left_distance" value="$(arg keep_lane_left_distance)"/>
    <param name="keep_lane_right_distance" value="$(arg keep_lane_right_distance)"/>
    <param name="max_boundingbox_side" value="$(arg max_boundingbox_side)"/>
    <param name="clip_min_height" value="$(arg clip_min_height)"/>
    <param name="clip_max_height" value="$(arg clip_max_height)"/>
    <param name="output_frame" value="$(arg output_frame)"/>
    <param name="remove_points_upto" value="$(arg remove_points_upto)"/>
    <param name="clustering_distance" value="$(arg clustering_distance)"/>
    <param name="cluster_merge_threshold" value="$(arg cluster_merge_threshold)"/>
    
    </node>

</launch>